pkg/core/resources/apis/mesh/dataplane_validator.go

```diff
@@ -20,12 +22,18 @@ func validateNetworking(networking *mesh_proto.Dataplane_Networking) validators.
 	if len(networking.GetInbound()) > 0 && networking.Gateway != nil {
 		err.AddViolationAt(path, "inbound cannot be defined both with gateway")
 	}
+	// backwards compatibility validate networking.address only when all inbounds are in new format
+	if !hasLegacyInbound(networking) && networking.Gateway == nil {
```

Comment:
What is the value in maintaining backwards compatibility if it leads to a situation where it's not possible to scrape metrics out of a `Dataplane` ?

Don't we want to prevent it from happening in the first place ?

@subnetmarco WDYT ?